src/data/matrix/rank.lean

```diff
@@ -52,19 +52,36 @@ lemma rank_le_width [strong_rank_condition R] {m n : ℕ} (A : matrix (fin m) (f
   A.rank ≤ n :=
 A.rank_le_card_width.trans $ (fintype.card_fin n).le
 
-lemma rank_mul_le [strong_rank_condition R] (A : matrix m n R) (B : matrix n o R) :
+lemma rank_mul_le_left [strong_rank_condition R] (A : matrix m n R) (B : matrix n o R) :
   (A ⬝ B).rank ≤ A.rank :=
 begin
   rw [rank, rank, to_lin'_mul],
   exact cardinal.to_nat_le_of_le_of_lt_aleph_0
     (rank_lt_aleph_0 _ _) (linear_map.rank_comp_le_left _ _),
 end
 
+include m_fin
+
+lemma rank_mul_le_right [strong_rank_condition R] (A : matrix l m R) (B : matrix m n R) :
+  (A ⬝ B).rank ≤ B.rank :=
+begin
+  classical,
```

Comment:
Why do we need `classical` here and not in the proof of `rank_mul_le_left`?

Reply:
The "another time" is now: #18800